spectator/registry.cc

```diff
@@ -26,7 +26,19 @@ auto Registry::GetLogger() const noexcept -> Registry::logger_ptr {
 }
 
 auto Registry::GetAgeGauge(Id id) noexcept -> std::shared_ptr<AgeGauge> {
-  return all_meters_.insert_age_gauge(std::move(id));
+  if (all_meters_.age_gauges_.contains(id)) {
+    return all_meters_.age_gauges_.at(id);
+  } else {
+    if (all_meters_.age_gauges_.size() < config_->age_gauge_limit) {
+      return all_meters_.insert_age_gauge(std::move(id));
+    } else {
+      logger_->warn("maximum number of age gauges (" +
+                    std::to_string(config_->age_gauge_limit) +
+                    ") has been reached, skipping creation. " +
+                    "change this value with the --age_gauge_limit flag.");
```

Comment:
It seems like the first step should be to ensure that the growth in age gauges is actually needed and not some sort of bug or issue. How would the user go about figuring out what is contributing to the number of gauges?

Reply:
I think that's a valid point. Maybe dump the Ids for the age gauges to the log file (once).

Reply:
In Atlas UI, filter down to all of the gauges reported from a known node that is experiencing this issue. If you know that the age gauge name can only be a handful of values and the tags are the source of the bug, then that provides another dimension for narrowing the search.

We briefly discussed adding some query capability for this use case, but it makes the daemon more complex with a questionable upside.

Reply:
I do like the idea of logging the age gauge ids once.

Reply:
I took a swing at this, logging each Id Name on a separate line. I considered duplicating the `id_to_string` function from `test_utils.cc` into the `meter_map`, so we would have a full accounting of the known Ids, but was not sure if we wanted to go this route.

Reply:
I have this working now with the existing Id formatter. It lists each known gauge by index and Id on a separate line, for readability.